Classes/Http/CorsHeaderMiddleware.php

```diff
@@ -14,6 +15,12 @@
 
 class CorsHeaderMiddleware implements MiddlewareInterface
 {
+    /**
+     * @Flow\Inject
+     * @var ResponseFactoryInterface
+     */
+    protected $responseFactory;
```

Comment:
Is there a specific reason that you used a @var annotation instead of typing the property?

Reply:
Just because it makes lazy injection easier. This made sense to me as this object can stay a proxy in all cases except OPTION requests.

I kinda dislike the code like this as it is harder to understand:

```
#[Flow\inject(lazy:true)]
protected ResponseFactoryInterface|LazyProxy|null $responseFactory;
```

I hope with PHP 8.4 we will be able to improve that eventually in Flow 10